modules/core/include/opencv2/core/cvdef.h

```diff
@@ -597,6 +597,13 @@ class float16_t
     #endif
     }
 
+    template<typename T>
+    float16_t& operator =(const T& val)
```

Comment:
There are two branches if `float16_t` implementation. They should have similar public interfaces.

Also conversion to float16 can be emulated and very slow, so we don't want to allow these automatic conversions for users - adding of this operator should be avoided.

FP16 support should be efficient.
There are no goals to push FP16 into all algorithms.